pkg/deployment/agent_test.go

```diff
@@ -158,6 +158,44 @@ func TestAgentLabels(t *testing.T) {
 	assert.Equal(t, fmt.Sprintf("%s-agent", a.jaeger.Name), dep.Spec.Template.Labels["app.kubernetes.io/name"])
 }
 
+func TestAgentDNSPolicyAndHostNetwork(t *testing.T) {
+	jaeger := v1.NewJaeger(types.NamespacedName{Name: "TestAgentDNSPolicyAndHostNetwork"})
+	jaeger.Spec.Agent.Strategy = "daemonset"
+	jaeger.Spec.Agent.DNSPolicy = "Default"
+	jaeger.Spec.Agent.HostNetwork = true
+	a := NewAgent(jaeger)
+	dep := a.Get()
+	assert.Equal(t, "Default", string(dep.Spec.Template.Spec.DNSPolicy))
+	assert.Equal(t, true, dep.Spec.Template.Spec.HostNetwork)
+
+}
+
+func TestAgentDNSPolicyAndHostNetwork1(t *testing.T) {
```

Comment:
The test needs a better name ;-) If it's just the same as the previous, with different values to test, you could use a table test.

https://github.com/golang/go/wiki/TableDrivenTests